File: leetcode/1509/MinDifference.js
let minDifference = (nums) => {
    if(nums.length <= 4) return 0;
    let min = Number.MAX_SAFE_INTEGER;
    
    nums.sort((a, b) => a - b);
    console.log(nums);
    for(let i = 0; i<=3; i++){
        console.log(nums.length-1-i,3-i);
        console.log(nums[nums.length-1-i],nums[3-i]);
        console.log('=========');
        min = Math.min(min,nums[nums.length-1-i] - nums[3-i]);
    }
    
    return min;
};

console.log(minDifference([1, 5, 0, 10, 14]));

File: leetcode/1509/MinDifference.c

#include <stdio.h>
#include <stdlib.h>

int cmpfunc(const void *a, const void *b)
{
    return (*(int *)a - *(int *)b);
}

int minDifference(int *nums, int numsSize)
{
    if (numsSize <= 4)
        return 0;
    int min = INT_MAX;

    // sort(nums, numsSize);
    qsort(nums, numsSize, sizeof(int), cmpfunc);

    for (int i = 0; i <= 3; i++)
    {
        min = min < nums[numsSize - 1 - i] - nums[3 - i] ? min : nums[numsSize - 1 - i] - nums[3 - i];
    }

    return min;
}